packages/runtime/src/useCases/consumption/requests/AcceptIncomingRequest.ts

```diff
@@ -20,6 +25,29 @@ export class AcceptIncomingRequestUseCase extends UseCase<AcceptIncomingRequestR
             return Result.fail(RuntimeErrors.general.recordNotFound(LocalRequest));
         }
 
+        if (
+            localRequest.source?.type === "RelationshipTemplate" &&
+            ![LocalRequestStatus.Decided, LocalRequestStatus.Completed, LocalRequestStatus.Expired].includes(localRequest.status)
+        ) {
+            const template = await this.relationshipTemplateController.getRelationshipTemplate(localRequest.source.reference);
+
+            if (!template) {
+                return Result.fail(RuntimeErrors.general.recordNotFound(RelationshipTemplate));
+            }
+
+            const existingRelationshipsToPeer = await this.relationshipController.getExistingRelationshipsToIdentity(localRequest.peer);
+
+            if (existingRelationshipsToPeer.length === 0 && template.cache?.expiresAt && template.isExpired()) {
+                await this.incomingRequestsController.updateRequestExpiryRegardingTemplate(localRequest, template.cache.expiresAt);
```

Comment:
> I'd prefer an update (including the expiryDate) when the request is received.

@Magnus-Kuhn How do you plan on doing that? Isn't the catch that the status of the Request is changed once the RelationshipTemplate reaches its expiration date? And we cannot do so in the moment when we receive the Request if the RelationshipTemplate isn't expired yet.

Reply:
@Milena-Czierlinski I meant  that only the expiryAt of the request is updated.
Also I just checked how it's currently done and GETting a request also updates the status upon expiry - so that makes the side effect ok even though I'm not sure if there is no way to get to the data without the get or if people may naively rely on a RequestStatusChanged event to Expired

Reply:
I now added the automatic creation of an expiration date of a Request in the onNewRelationship property of a RelationshipTemplate if none is provided on the creation of a RelationshipTemplate. Adding this also to receiving a RelationshipTemplate is a bit more tricky, since given the current implementation we only have the information of whether the Request is from the onNewRelationship or onExistingRelationship property on a high level. So this would be a bigger effort and I'm not sure yet if it's worth it. @Magnus-Kuhn What do you say?

Reply:
we can check if there is a relationship in `received` of the `IncomingRequestsController`